src/components/interactive-map.js

```diff
@@ -362,6 +372,13 @@ const InteractiveMap = forwardRef((props, ref) => {
     };
   }, []);
 
+  useLayoutEffect(() => {
```

Comment:
based on [this](https://reactjs.org/docs/hooks-reference.html#uselayouteffect), `useLayoutEffect` offers a sync side effect and it may block the painting process. Is this the expected behavior?

Reply:
Yes. The reason the controller needs to call `onViewportChange` is to intercept any viewport change before it's rendered. An example:

- Current viewport - `longitude: 0, latitude: 0, zoom: 10`
- User set props - `longitude: 0, latitude: 0, zoom: 6, transitionInterpolator: new LinearInterpolator(), transitionDuration: 1000`
- Controller triggers transition, calls `onViewportChange` - `longitude: 0, latitude: 0, zoom: 9.9`
- Updates map - `longitude: 0, latitude: 0, zoom: 9.9`

Note here we don't want `zoom: 6` to render before `zoom: 9.9`.